src/bin/lib/subcommands.c/runprogram.h

```diff
@@ -77,51 +80,55 @@ static void waitprogram(Program *prog, pid_t childPid);
  * the run and then return a Program struct instance with the result of running
  * the program.
  */
-Program
+Program *
 run_program(const char *program, ...)
 {
 	int nb_args = 0;
 	va_list args;
 	const char *param;
-	Program prog = { 0 };
-
-	prog.program = strdup(program);
-	prog.returnCode = -1;
-	prog.error = 0;
-	prog.setsid = false;
-	prog.capture = true;
-	prog.tty = false;
-	prog.processBuffer = NULL;
-	prog.stdOutFd = -1;
-	prog.stdErrFd = -1;
-	prog.stdOut = NULL;
-	prog.stdErr = NULL;
-
-	prog.args = (char **) malloc(ARGS_INCREMENT * sizeof(char *));
-	prog.args[nb_args++] = prog.program;
+	Program *prog = (Program *) malloc(sizeof(Program));
+
+	if (prog == NULL)
+	{
+		return NULL;
```

Comment:
Callers of this function does not check for NULL values, and may create segfaults.

Reply:
Added NULL checks in the caller. The subtlety here is that because of the core organization (we vendor-in an external lib) we can't `log_error` from `lib/subcommands.c/*`.